pydb/services.py

```diff
@@ -89,23 +80,33 @@ def as_dict_for_monkey_patching_old_psutils(self, attrs=[], ad_value=None):
                 name = 'cwd'
         retdict[name] = ret
     return retdict
-                                                                                                                                                                                                                                                                                                
+
+
 if not hasattr(psutil.Process, 'as_dict'):
     psutil.Process.as_dict = as_dict_for_monkey_patching_old_psutils
 
 
+def maybe_parent_service_process(service_process):
+    try:
+        while executionenvironment.is_montag_process(
+                service_process.parent().as_dict(attrs=['name', 'username', 'cmdline']), names):
+            service_process = service_process.parent()
+    except psutil.AccessDenied:
+        pass
+
+    return service_process
+
+
 def get_current_services_status():
     services_status = _get_default_services_status()
-    for pid in pidlist():
-        try:
-            p = psutil.Process(pid)
-            pinfo = p.as_dict(attrs=['name', 'exe', 'username', 'status', 'cmdline'])
-
-            detected_service_name = executionenvironment.is_montag_process(pinfo, names)
+    try:
+        for p in psutil.process_iter(attrs=['name', 'exe', 'username', 'status', 'cmdline']):
```

Comment:
At least on ubuntu 14.04 (which is quite old, I know) this does not work - the iter command there takes no arguments.
Ubuntu 14.04 uses psutil 1.2.1.

Reply:
On the other hand, ubuntu 14.04 is reaching EOL just now :)

Reply:
I'm voting for the `alte Zöpfe abschneiden` here.

Reply:
we should also remove the monkey patch for as_dict on the way